lisp/embed.go

```diff
@@ -142,16 +142,11 @@ func GoMap(v *LVal) (map[interface{}]interface{}, bool) {
 	if v.Type != LSortMap {
 		return nil, false
 	}
-	lmap := v.Map()
-	m := make(map[interface{}]interface{}, len(lmap))
-	for k, vlisp := range lmap {
-		vgo := GoValue(vlisp)
-		switch k.(type) {
-		case MapSymbol:
-			m[string(k.(MapSymbol))] = vgo
-		default:
-			m[k] = vgo
-		}
+	data := v.Map()
+	m := make(map[interface{}]interface{}, data.Len())
+	for _, pair := range sortedMapEntries(data).Cells {
+		// FIXME:  typecheck pair
+		m[GoValue(pair.Cells[0])] = GoValue(pair.Cells[1])
```

Comment:
what's the consequence of not checking the type here? it only effects embedded use cases correct?

Reply:
Well, if you assume that `pair.Cells[0]` could be *any* type then it would be possible for this to panic. Because GoValue can return things like `[]interface{}`, `map[string]interface{}`, `error` and potentially other things which are not valid map keys.

For the existing sorted-map implementations this can't actually panic because `GoValue(pair.Cells[0])` will always return a string. So if someone wanted to extend their application so that sorted-maps could use vectors as keys (or something like that) then this probably wouldn't be sufficient.

IDK. I guess I could at least elaborate on what risks this poses. This function signature doesn't have a lot of room for signaling an failure because of bad map keys... I guess you document that returning `nil, true`  means the map had invalid keys. That seems a bit random but I guess it could work.

There's a similar FIXME comment in lisp.go... I might be able to do some better error signalling there.